src/BinOp.chpl

```diff
@@ -261,6 +261,8 @@ module BinOp
       var repMsg = "created %s".format(st.attrib(rname));
       return new MsgTuple(repMsg, MsgType.NORMAL);
     }
-    return new MsgTuple("Bin op not supported", MsgType.NORMAL);
+    var errorMsg = notImplementedError(pn,l.dtype,op,r.dtype);
+    omLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+    return new MsgTuple(errorMsg, MsgType.ERROR);
```

Comment:
> I'd like @bmcdonald3 to weigh in on if the notImplementedError seems appropriate

I am happy with this; to be honest, the error I was throwing was created only out of laziness and should have been like this from the start, so thank you for dealing with my laziness!